### PR TITLE
Support running SCT with scylla-cloud

### PR DESCRIPTION
**Pull request by fruch**
example of needed config to use it
```yaml

db_type: cloud_scylla
n_monitor_nodes: 0
ip_ssh_connections: public

cloud_credentials_path: '~/.ssh/support'
authenticator: PasswordAuthenticator
authenticator_user: scylla
authenticator_password: kqmiwI8czN1d5rb


n_db_nodes: 6
db_nodes_public_ip: [ "63.35.208.180", "34.253.248.222", "99.80.52.252", "34.246.3.254", "18.203.22.186", "99.80.68.172" ]
db_nodes_private_ip:  [ "10.0.0.11", "10.0.1.11", "10.0.2.11", "10.0.0.12","10.0.1.12", "10.0.2.12" ]

```

### CODE REVIEW

File: sdcm/cluster_baremetal.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def node_setup(self, node, verbose=False):
         We have to modify scylla.yaml on our own because we are not on AWS,
         where there are auto config scripts in place.
         """
-        self._node_setup(node, verbose)
+        pass  # self._node_setup(node, verbose)
```
Comment (roydahan) [inline sdcm/cluster_baremetal.py]: Is this intended?
Won't it affect our support of bare metal?

Comment (fruch) [reply sdcm/cluster_baremetal.py]: it's intended, bare metal doesn't have this function... looks like it's actually fixes baremetal support...
We'll need to revisit baremetal support, I don't think it was used in a while...

File: sdcm/cluster.py [hunk 2]
```diff
@@ -2373,6 +2378,11 @@ def get_loader(self):
             self._loader_queue = [node for node in self.nodes]
         return self._loader_queue.pop(0)
 
+    def get_cql_auth(self):
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster.py]: they are the same on two different classes, it's reading the same values, we can consolidate it into one function out of those classes

**Comment by fruch** [comment]
@roydahan 
here is a run I've left for 12 hours:
http://jenkins.cloudius-systems.com:8080/view/scylla-3.0/job/scylla-3.0/job/3.0-test/job/scylla-longevity-test(new-config)/38/label=aws-scylla-qa-builder4/console

File: sdcm/cluster.py [hunk 3]
```diff
@@ -64,7 +64,7 @@
 TEST_DURATION = 60
 # max limit of coredump file can be uploaded(5 GB)
 COREDUMP_MAX_SIZE = 1024 * 1024 * 1024 * 5
-IP_SSH_CONNECTIONS = 'public'
+IP_SSH_CONNECTIONS = 'private'
```
Comment (bentsi) [inline sdcm/cluster.py]: dangerous :)

File: sdcm/cluster.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bentsi) [reply sdcm/cluster.py]: put it in the base class

File: sdcm/cluster_aws.py [hunk 4]
```diff
@@ -188,7 +188,7 @@ def _create_spot_instances(self, count,  interfaces, ec2_user_data='', dc_idx=0,
                 if ec2_client.MAX_SPOT_EXCEEDED_ERROR in cl_ex.message:
                     self.log.debug('Cannot create spot instance(-s): %s.'
                                    'Creating on demand instance(-s) instead.', cl_ex)
-                    instances_i = self._create_on_demand_instances(count, interfaces, ec2_user_data, dc_idx)
+                    instances_i = self._create_on_demand_instances(count, interfaces, ec2_user_data, dc_idx, tags_list=tags_list)
```
Comment (bentsi) [inline sdcm/cluster_aws.py]: I think we decided not to fallback to on_demand. Right @roydahan ?

File: sdcm/cluster_baremetal.py [hunk 5]
```diff
@@ -91,10 +91,10 @@ def add_nodes(self, count, dc_idx=0):
     def destroy(self):
         logger.info('Destroy nodes')
         for node in self.nodes:
-            node.destroy(force=True)
+            node.destroy()
```
Comment (bentsi) [inline sdcm/cluster_baremetal.py]: why did you remove the `force` arg?

File: sdcm/cluster.py (hunk 3 above)
Comment (roydahan) [reply sdcm/cluster.py]: I saw it too, but I think it's better like this.

File: sdcm/nemesis.py [hunk 6]
```diff
@@ -1059,6 +1066,19 @@ def disrupt(self):
...
+
+    @log_time_elapsed_and_status
+    def disrupt(self):
+        # Limit the nemesis scope to only one relevant to scylla cloud, where we defined we don't have AWS api access:
+        self.call_random_disrupt_method(disrupt_methods=['disrupt_nodetool_cleanup',
+                                                         'disrupt_nodetool_drain', 'disrupt_nodetool_refresh',
+                                                         'disrupt_major_compaction',
+                                                         'disrupt_modify_table', 'disrupt_nodetool_enospc',
+                                                         'disrupt_soft_reboot_node',
+                                                         'disrupt_truncate'])
```
Comment (roydahan) [inline sdcm/nemesis.py]: Add these nemesis to the list:
- StopWaitStartMonkey
- StopStartMonkey

Comment (roydahan) [reply sdcm/nemesis.py]: I see you have the second disrupt_stop_start_scylla_server

**Comment by roydahan** [comment]
@fruch, you are fast :)

**Comment by fruch** [comment]
@roydahan they call me Berry Alan (the Flash)

File: sdcm/cluster_baremetal.py (hunk 5 above)
Comment (fruch) [reply sdcm/cluster_baremetal.py]: it's doing nothing for the baremetal

Comment (fruch) [reply sdcm/cluster_baremetal.py]: it's not being used in baremetal

File: sdcm/nemesis.py [hunk 7]
```diff
@@ -80,10 +80,13 @@ def update_stats(self, disrupt, status=True, data={}):
             self.tester.update({'nemesis': self.stats})
 
     def set_target_node(self):
-        non_seed_nodes = [node for node in self.cluster.nodes if not node.is_seed]
-        # if non_seed_nodes is empty, nemesis failed. 
-        # TODO need checks
-        self.target_node = random.choice(non_seed_nodes)
+        run_only_on_seed = self.cluster.params.get('nemesis_only_seeds', default=True)
```
Comment (bentsi) [inline sdcm/nemesis.py]: change name

File: sdcm/nemesis.py [hunk 8]
```diff
@@ -1059,6 +1066,20 @@ def disrupt(self):
                                                          'disrupt_restart_then_repair_node', 'disrupt_truncate'])
 
 
+class CloudLimitedChaosMonkey(Nemesis):
```
Comment (bentsi) [inline sdcm/nemesis.py]: ScyllaCloud...

File: sdcm/sct_config.py [hunk 9]
```diff
@@ -117,7 +117,10 @@ class SCTConfiguration(dict):
         dict(name="user_credentials_path", env="SCT_USER_CREDENTIALS_PATH", default='~/.ssh/scylla-test', type=str, required=True,
              help="""Path to your user credentials. qa key are downloaded automatically from S3 bucket"""),
 
-        dict(name="ip_ssh_connections", env="SCT_IP_SSH_CONNECTIONS", default='public', type=str, required=False,
+        dict(name="cloud_credentials_path", env="SCT_CLOUD_CREDENTIALS_PATH", default='~/.ssh/scylla-test', type=str, required=True,
```
Comment (bentsi) [inline sdcm/sct_config.py]: default `~/.ssh/support`

File: sdcm/sct_config.py [hunk 10]
```diff
@@ -275,6 +284,8 @@ class SCTConfiguration(dict):
                  This value is supposed to reproduce
                  https://github.com/scylladb/scylla/issues/1140
              """),
+        dict(name="nemesis_only_seeds", env="SCT_NEMESIS_ONLY_SEEDS", default=True, type=boolean, required=False,
```
Comment (bentsi) [inline sdcm/sct_config.py]: change accordingly to above

File: sdcm/sct_config.py [hunk 11]
```diff
@@ -722,7 +733,10 @@ class SCTConfiguration(dict):
         'openstack': ['openstack_user', 'openstack_password', 'openstack_tenant', 'openstack_auth_version',
                       'openstack_auth_url', 'openstack_service_type', 'openstack_service_name', 'openstack_service_region',
                       'openstack_instance_type_loader', 'openstack_instance_type_db', 'openstack_instance_type_monitor',
-                      'openstack_image', 'openstack_image_username', 'openstack_network']
+                      'openstack_image', 'openstack_image_username', 'openstack_network'],
+        'aws-siren': ["user_prefix", "instance_type_loader", "region_name", "security_group_ids", "subnet_id",
```
Comment (bentsi) [inline sdcm/sct_config.py]: add filter seeds here

File: sdcm/tester.py [hunk 12]
```diff
@@ -165,6 +165,7 @@ def __init__(self, methodName='test', name=None, params=None,
         cluster.Setup.set_test_id(self.params.get('test_id'))
         cluster.Setup.reuse_cluster(self.params.get('reuse_cluster', default=False))
         cluster.Setup.keep_cluster(self._failure_post_behavior)
+        cluster.Setup.set_multi_region(self.params.get('region_name').split() > 1)
```
Comment (bentsi) [inline sdcm/tester.py]: :+1:

File: test-cases/longevity-200GB-48h-verifier-LimitedMonkey-tls.yaml [hunk 13]
```diff
@@ -0,0 +1,33 @@
+test_duration: 2880
+prepare_write_cmd: "cassandra-stress write cl=ALL n=200200300  -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=1000 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=15"
+prepare_verify_cmd: "cassandra-stress read cl=ALL n=200200300  -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=2000 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=15"
+stress_cmd: ["cassandra-stress write cl=QUORUM duration=2860m  -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=250  -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=400200300..600200300 -log interval=15"]
+stress_read_cmd: ["cassandra-stress read cl=ONE duration=2860m -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=250  -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=5"]
+run_fullscan: 'keyspace1.standard1, 5' # 'ks.cf|random, interval(min)''
+
```
Comment (bentsi) [inline test-cases/longevity-200GB-48h-verifier-LimitedMonkey-tls.yaml]: Please add `ssh_remote: "RemoteFabric"` to start using it

Comment (fruch) [reply test-cases/longevity-200GB-48h-verifier-LimitedMonkey-tls.yaml]: I would add it to the jenkins job, this parameter doesn't belong in the testcases
once we thing it's good enough, we'll change it to be the default.

**Comment by bentsi** [comment]
unable to backport to others, too many conflicts

**Comment by roydahan** [comment]
Of course we don’t need to backport it to 3.0 and 2018.1 - only critical fixes are going in for these branches.

2019.1 is ok to backport
